web_widget_image_webcam/static/src/js/webcam_widget.js

```diff
@@ -9,129 +9,124 @@ odoo.define("web_widget_image_webcam.webcam_widget", function (require) {
     var core = require("web.core");
     var rpc = require("web.rpc");
     var Dialog = require("web.Dialog");
-    var FieldBinaryImage = require("web.basic_fields").FieldBinaryImage;
+    const {patch} = require("@web/core/utils/patch");
+    const {ImageField} = require("@web/views/fields/image/image_field");
 
     var _t = core._t;
     var QWeb = core.qweb;
 
-    FieldBinaryImage.include({
-        _render: function () {
-            this._super();
-
-            var self = this,
-                WebCamDialog = $(QWeb.render("WebCamDialog")),
-                img_data = false;
+    patch(ImageField.prototype, "web_widget_image_webcam", {
+        setup() {
+            var self = this;
+            self._super.apply(self, arguments);
+            self.flash_fallback_mode = false;
+            rpc.query({
+                model: "ir.config_parameter",
+                method: "get_webcam_flash_fallback_mode_config",
+            }).then(function (default_flash_fallback_mode) {
+                self.flash_fallback_mode = default_flash_fallback_mode;
+            });
```

Comment:
I'd put this rpc query outside the setup method. Simply put it in the global scope.
Because we don't need to read this value from the server for each field widget instance, we only need to read it once.

You can find an example here: https://github.com/OCA/web/blob/0253aa72c554cbaf68080725dd091ce74162bf9a/web_widget_ckeditor/static/src/js/field_ckeditor.js#L20

Anyway, not blocking as it was like this before the migration

Reply:
Hi @ivantodorovich ,

thank you for your review.

I'm not sure about your suggestion; I mean, the global variables should be set after browser reload, so when assets are loaded. if parameter is changed global variable will not be updated, there is something I'm missing about that?

Reply:
Well, reading this setting requires 1 HTTP request, so it's better not to abuse it.
If you don't expect the setting to change often (and you don't on this case), then a browser reload is more than acceptable

Reply:
ok thanks. based on this I'll fix that too.